### DISCUSSION TITLE
Using section id instead of heading id for sub results

### DISCUSSION THREAD

**Discussion by JorisHeyse** [Pagefind/pagefind#518] [Ideas]
Thank you for the great work with this search library! I'm very pleased with its ease of integration and good performance out of the box.

I recently integrated pagefind for our documentation. Unfortunately, I'm having an issue with sub results. I read that sub results are created for each heading that has an `id` attribute. My headings don't have an `id` but the `section` tags wrapped around them do have an `id`. We're using Sphinx to build our documentation with the ReadTheDocs theme, so making changes to the generated HTML is not really an option for us. It would be nice if the indexer could check for an `id` on the parent `section` tag in case a heading doesn't have one. 
This would open up the possibility to show sub results in our search as well. As we have quite a large documentation, it could greatly improve the search experience.

Looking forward to hearing your thoughts on this.

**Comment by bglw** [comment]
Apologies, missed this!

I'll have a think for this — it definitely seems helpful to have, and in fact the search result data already contains every `id` attribute on the page, heading or not.

The hard part will be determining what text to show for that sub result, as it isn't trivially paired with the heading element to capture the text. 

Will have a ponder and get back to you 🙂

**Comment by JorisHeyse** [comment]
No problem! 

I think I understand the difficulty. 
In Sphinx, a section is created for each heading so the first element in the section is normally a heading. A subheading then creates a section within that section object. This reflects how Sphinx builds doctrees from .rst files. Of course this may be different for other doc generation engines and I guess therein lies the problem.

I appreciate your consideration of this.

**Comment by JorisHeyse** [comment]
Hi, Hope you're doing well.
We're still very happy with the pagefind library. It greatly increased the searchability of our documentation. 

I was wondering if you've had time to consider my request above, we would really love to have these sub results for our search queries.
